### PR TITLE
Addon-knobs: : add number config options

### PR DESCRIPTION
**Pull request by lifeiscontent**
Fix inconsistency in number API

fix step,min,max attributes in number type

Issue:

## What I did

Fixed inconsistency in number type.

## How to test

when setting the options of a number, the step,min,max attributes now get applied.

Is this testable with jest or storyshots? Yes, but I didn't write a test.

Does this need a new example in the kitchen sink apps? No.

Does this need an update to the documentation? Not really, but I could add an example.

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by Hypnosphi** [comment]
> Does this need a new example in the kitchen sink apps

Actually, I think it does. Please add one here: https://github.com/storybooks/storybook/blob/master/examples/cra-kitchen-sink/src/stories/addon-knobs.stories.js

You may need to run `yarn jest -u` to update storyshots after that

**Comment by ndelangen** [comment]
Thank you for this PR @lifeiscontent !

@Hypnosphi Looks like the min,max,step was already in the example?
https://github.com/storybooks/storybook/blob/master/examples/cra-kitchen-sink/src/stories/addon-knobs.stories.js#L52

**Comment by Hypnosphi** [comment]
@ndelangen we need the same but without `range: true`

**Comment by lifeiscontent** [comment]
@Hypnosphi how do I run the kitchen sink example?

I tried the following.

```
yarn
yarn bootstrap # and selected core and examples
cd storybook/examples/cra-kitchen-sink
yarn storybook
```

but I get issues saying none of the storybook modules exist.

**Comment by lifeiscontent** [comment]
@Hypnosphi updated!

File: examples/cra-kitchen-sink/src/stories/addon-knobs.stories.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ class AsyncItemLoader extends React.Component {
 
 storiesOf('Addon Knobs.withKnobs', module)
   .addDecorator(withKnobs)
-  .add('tweaks static values', () => {
+  .add('tweaks static values via range', () => {
```
Comment (Hypnosphi) [inline examples/cra-kitchen-sink/src/stories/addon-knobs.stories.js]: This example includes a lot of other knobs

Comment (Hypnosphi) [reply examples/cra-kitchen-sink/src/stories/addon-knobs.stories.js]: Let’s add min/max/step to `dollars` below instead of duplicating the example

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

File: examples/cra-kitchen-sink/src/stories/addon-knobs.stories.js (hunk 1 above)
Comment (Hypnosphi) [reply examples/cra-kitchen-sink/src/stories/addon-knobs.stories.js]: Please run `yarn jest -u` after that

**Comment by lifeiscontent** [comment]
@Hypnosphi updated.

**Comment by Hypnosphi** [comment]
Ok looks like it passes as is. I'll be to test the example manually in a few hours

**Review by Hypnosphi — APPROVED** [review]

**Pull request merged**